src/js/media-error-dialog.ts

```diff
@@ -0,0 +1,108 @@
+import { globalThis } from './utils/server-safe-globals.js';
+import { MediaUIAttributes } from './constants.js';
+import { errors } from './labels/labels.js';
+import { MediaChromeDialog } from './media-chrome-dialog.js';
+import {
+  getNumericAttr,
+  getStringAttr,
+  setNumericAttr,
+  setStringAttr,
+} from './utils/element-utils.js';
+
+function getSlotTemplateHTML(attrs: Record<string, string>) {
+  return /*html*/ `
+    <style>
+      #content {
+        background: var(--media-control-background, var(--media-secondary-color, rgb(20 20 30 / .8)));
+        display: block;
+        padding: 1.2em 1.5em;
+      }
+
+      h3,
+      p,
+      ::slotted(h3),
+      ::slotted(p) {
+        margin-block: 0 .3em;
+      }
+    </style>
+    <slot name="error-${attrs.mediaerrorcode}" id="content">
+      ${getErrorMessage(attrs.mediaerrorcode, attrs.mediaerrormessage)}
+    </slot>
+  `;
+}
+
+function shouldHideErrorDialog(errorCode?: number) {
+  return !errorCode || errors[errorCode] === null;
+}
+
+function getErrorMessage(errorCode?: number | string, errorMessage?: string) {
+  const message: string = errors[+errorCode] ?? errorMessage ?? '';
```

Comment:
question: does this mean we have to provide a message in the format `Title:Verbose Message` for a title to be used? Also, if we treat `errors[+errorCode]` as the preference, doesn't that mean folks modeling more verbose/context specific error messages will be ignored for the more generic ones in this case?

Reply:
yes, the message will need a colon for a title to be used.

yes, my first thought was that it's a nicer experience to have consistent error messages across browsers because they're different. 

it's a valid concern, it is possible to clear the errors labels and then it will fallback to the message in the error object but that's not up to the custom media element. we could potentially make use of detecting if the error event is a native event or user event (isTrusted)?

Reply:
I'm thinking particularly about the far more verbose/"bespoke" error messages and titles in `mux-player`, fwiw

Reply:
the `isTrusted` is an interesting solution... I think I like it (independent of whether this should be handled as a function vs. a lookup).